source/rpt/rpt_fem_reconstruction.cc

```diff
@@ -0,0 +1,1080 @@
+          // Check if the new cost is lower than the previously stored
+          // real_location's
+          if (calculated_cost < max_cost_function)
+            {
+              position_found         = true;
+              max_cost_function      = calculated_cost;
+              previous_position_cell = adjacent_cell;
+
+              // Evaluate the real location of the particle
+              for (unsigned int v = 0; v < adjacent_cell->n_vertices(); ++v)
+                {
+                  if (v == 0)
+                    real_location = adjacent_cell->vertex(v);
+                  else
+                    real_location +=
+                      reference_location[v - 1] *
+                      (adjacent_cell->vertex(v) - adjacent_cell->vertex(0));
+                }
+            }
+        }
+    }
+
+  if (position_found)
+    {
+      found_positions.push_back(real_location);
+      return true;
+    }
+  else
+    return false;
+}
+
+template <int dim>
+void
+RPTFEMReconstruction<dim>::read_experimental_counts(
+  std::vector<std::vector<double>> &all_experimental_counts)
+{
+  TimerOutput::Scope t(computing_timer, "read_experimental_counts");
+
+  std::ifstream in(
+    rpt_parameters.fem_reconstruction_param.experimental_counts_file);
+  double value;
+
+  if (in)
+    {
+      std::string line;
+
+      while (std::getline(in, line))
+        {
+          // Ignore empty lines
+          if (line == "")
+            continue;
+
+          all_experimental_counts.emplace_back(std::vector<double>());
+
+          // Break down the row into column values
+          std::stringstream split(line);
+
+          while (split >> value)
+            all_experimental_counts.back().push_back(value);
+        }
+    }
+}
+
+
+template <int dim>
+void
+RPTFEMReconstruction<dim>::trajectory()
+{
+  // Tolerance or the extrapolation limit in the reference space for a found
+  // position
+  double tol_reference_location = 0.005;
+
+  if (rpt_parameters.fem_reconstruction_param.mesh_type ==
+      Parameters::RPTFEMReconstructionParameters::FEMMeshType::dealii)
+    {
+      const unsigned int power =
+        pow(2, rpt_parameters.fem_reconstruction_param.mesh_refinement);
+      const unsigned int n_cell_z =
+        2 * rpt_parameters.fem_reconstruction_param.z_subdivisions * power;
+      tol_reference_location =
+        rpt_parameters.rpt_param.reactor_height / n_cell_z * 1.15;
+    }
+
+  std::cout << "tol: " << tol_reference_location << std::endl;
+
+  // Read and store all experimental counts
+  std::vector<std::vector<double>> all_experimental_counts;
+  read_experimental_counts(all_experimental_counts);
+
+  {
+    TimerOutput::Scope t(computing_timer, "find_particle_positions");
+
+    if (rpt_parameters.fem_reconstruction_param.search_type ==
+        Parameters::RPTFEMReconstructionParameters::FEMSearchType::local)
+      {
+        bool adjacent_cell_search = false;
```

Comment:
Hey Amishga, 
Maybe you can add a comment here and explain why you start with global search at the beginning.